pages/_app.js

```diff
@@ -82,15 +123,23 @@ export default function App({ Component, pageProps }) {
     );
   }
 
+  useEffect(() => {
+    const count =
+      filters.fertiliserSeason.length +
+      (filters.lightNeed ? 1 : 0) +
+      (filters.waterNeed ? 1 : 0);
+    setFilterCount(count);
+  }, [filters]);
+
```

Comment:
Not 100% sure, but I guess you don't need the `useEffect` 🧐

- `filterCount` is not an independent state, but can be calculated/derived directly from the `filters` at any time.

➡️ If the `filterCount` is calculated dynamically, you don't need to use `useEffect` and set the state separately.

Example:
```js

function calculateFilterCount(filters) {
  return (
    filters.fertiliserSeason.length +
    (filters.lightNeed ? 1 : 0) +
    (filters.waterNeed ? 1 : 0)
  );
}

// Instead of the useEffect:
const filterCount = calculateFilterCount(filters);

```